### ISSUE TITLE
Shade Animation Window Effect

### ISSUE THREAD

**Issue by harogaston** [linuxmint/Cinnamon#3819]
I think the title is quite selfexplanatory. Among the different effects under the 'Effects' section in 'System Settings' it would be nice to have a simple roll-up shade animation similar to compiz.
I am having a look at the code, and will see what I can do, currently looking into this file: "Cinnamon/files/usr/lib/cinnamon-settings/modules/cs_effects.py"

Any help or ideas will be appreciated.

Thank you

**Comment by sprintcowboy** [comment]
Is this what you are talking about?
System settings - windows - under Titlebar , Action on title bar double-click, middle-click, or right-click offers toggle shade.

**Comment by harogaston** [comment]
Hi,
No, it is not that. I am aware that shading is possible in cinnamon. I would like to have a roll-up like animation for that. As there are animations for minimizing, maximizing and closing windows. I am talking about what is in the 'Effects' section.

**Comment by mtwebster** [comment]
This is a neat idea - If you're interested, you'll have to implement this in both Cinnamon and muffin - you can look at these commits, from where I implemented the tiling/snapping animations not too long ago:

https://github.com/linuxmint/Cinnamon/commit/0c6bc3e6adc50f495fabc44ec0ff067acee709ed
https://github.com/linuxmint/muffin/commit/e06ac1b26d3dfb06d321e505eccf1a037535b627

I'll add this also to our internal 'wish list'

**Comment by JosephMcc** [comment]
I actually have this mostly done. Waiting on some other PR's to get in and then I'll do a proper one for this.

**Comment by harogaston** [comment]
Wow! That's actually great. Looking forward to it. Can you provide a link to your code?
Thanks.

**Comment by anandrkris** [comment]
@JosephMcc - Is this implemented?

**Comment by JosephMcc** [comment]
No. I had it done at one point and then lost it with a poor backup routine :(
I'll try and revisit it at some point.

**Comment by harogaston** [comment]
I will close this, as I am not longer interested.
